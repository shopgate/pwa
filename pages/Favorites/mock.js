const foo = {
  productData: {
    name: 'Foo',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'foo',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPrice: 101,
      currency: 'EUR',
    },
  },
};

const bar = {
  productData: {
    name: 'Bar',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'bar',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPrice: 101,
      currency: 'EUR',
    },
  },
};

export const mockedEmptyState = {
  product: {
    productsById: {
      foo,
    },
  },
  favorites: {
    products: {
      ids: [],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedState = {
  ...mockedEmptyState,
  favorites: {
    products: {
      ids: ['foo'],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedNotReadyState = {
  ...mockedEmptyState,
  favorites: {
    products: {},
    isFetching: false,
  },
};

export const mockedNextProps = {
  products: [
    foo.productData,
    bar.productData,
  ],
};

