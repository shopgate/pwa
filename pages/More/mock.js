import configureStore from 'redux-mock-store';

const defaultState = {
  client: {
    isFetching: false,
    appVersion: '0-app',
    libVersion: '0-lib',
    codeBaseVersion: '0-code',
  },
  menu: {
    menusById: {
      quickLinks: {
        entries: [
          {
            url: '/page/test',
            label: 'CMS Test',
          },
        ],
      },
    },
  },
};

const userLoggedInState = {
  ...defaultState,
  user: {
    login: {
      isLoggedIn: true,
    },
    data: {
      firstName: 'John',
      lastName: 'Doe',
    },
  },
};

const userLoggedOutState = {
  ...defaultState,
  user: {
    login: {
      isLoggedIn: false,
    },
    data: {},
  },
};

/**
 * Mocked store.
 * @returns {Object}
 */
export const getLoggedInStore = () => configureStore()(userLoggedInState);

/**
 * Mocked store.
 * @returns {Object}
 */
export const getLoggedOutStore = () => configureStore()(userLoggedOutState);
