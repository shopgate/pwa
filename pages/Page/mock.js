const widgetsPageRoute = {
  id: 'd82ea770-5627-4d1e-a301-a06bbe0d5cbd',
  params: {
    pageId: 'cms_test',
  },
  pathname: '/page/cms_test',
  pattern: '/page/:pageId',
  query: {},
  state: {},
};

const homeRoute = {
  id: 'a4bf3a32-0ed8-4138-bb94-098f32f3aeb3',
  params: {},
  pathname: '/',
  pattern: '/',
  query: {},
  state: {},
};

const page = {
  cms_test: {
    title: 'CMS Test Page',
    widgets: [
      {
        type: '@shopgate/commerce-widgets/html',
        settings: {
          html: 'Hi Test test',
          latestEditorMode: 'source',
          defaultPadding: true,
        },
        id: 'cms_test-0-@shopgate/commerce-widgets/html',
      },
    ],
    isFetching: false,
    expires: 9999999999999,
  },
};

const routerState = {
  router: {
    routing: false,
    stack: [
      { ...homeRoute },
      { ...widgetsPageRoute },
    ],
  },
};

const widgetsInitialState = {
  page: {},
  ui: {
    general: {
      title: null,
    },
  },
  ...routerState,
};

const widgetsState = {
  page,
  ui: {
    general: {
      title: null,
    },
  },
  ...routerState,
};

export {
  widgetsState,
  widgetsInitialState,
  widgetsPageRoute,
};
