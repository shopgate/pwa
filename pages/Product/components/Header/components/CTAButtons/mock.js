export const mockedState = {
  product: {
    currentProduct: {
      productId: 'product_1',
    },
    productsById: {
      product_1: {
        isFetching: false,
        productData: {
          id: 'product_1',
          name: 'Product No. 1',
          flags: {
            hasVariants: true,
            hasOptions: false,
          },
          stock: {
            orderable: true,
          },
        },
      },
      product_2: {
        isFetching: false,
        productData: {
          id: 'product_2',
          baseProductId: 'product_1',
          name: 'Product No. 2',
          flags: {
            hasVariants: false,
            hasOptions: false,
          },
        },
      },
    },
  },
  favorites: {
    products: {
      isFetching: false,
      ids: ['product_1', 'product_2'],
    },
  },
  cart: {
    productPendingCount: 0,
  },
};

export const mockedVariantState = {
  product: {
    currentProduct: {
      productId: 'product_1',
      productVariantId: 'product_2',
    },
    productsById: {
      product_1: {
        isFetching: false,
        productData: {
          id: 'product_1',
          name: 'Product No. 1',
          flags: {
            hasVariants: true,
            hasOptions: false,
          },
          stock: {},
        },
      },
      product_2: {
        isFetching: false,
        productData: {
          id: 'product_2',
          baseProductId: 'product_1',
          name: 'Product No. 2',
          flags: {
            hasVariants: false,
            hasOptions: false,
          },
          stock: {
            orderable: true,
          },
        },
      },
    },
    optionsByProductId: {
      product_1: null,
      product_2: null,
    },
  },
  favorites: {
    products: {
      isFetching: false,
      ids: ['product_1', 'product_2'],
    },
  },
  cart: {
    productPendingCount: 0,
  },
};
