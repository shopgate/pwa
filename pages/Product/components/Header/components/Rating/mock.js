/**
 * Mock for getElementById
 * @param {function} scrollSpy Element.scroll spy function.
 * @returns {function}
 */
const getElementById = scrollSpy => () => ({
  offsetTop: 100,
  closest() {
    return {
      scroll: scrollSpy,
    };
  },
});

export { getElementById };
