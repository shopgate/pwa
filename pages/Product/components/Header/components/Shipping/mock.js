export const mockedStoreWithShippingPrice = {
  product: {
    shippingByProductId: {
      fakeId: {
        shipping: {
          price: 1,
          currency: 'EUR',
        },
      },
    },
  },
};
export const mockedStoreWithFreeShipping = {
  product: {
    ...mockedStoreWithShippingPrice.product,
    shippingByProductId: {
      fakeId: {
        shipping: {
          price: 0,
          currency: 'EUR',
        },
      },
    },
  },
};

export const mockedStoreWithUnknownShipping = {
  product: {
    ...mockedStoreWithShippingPrice.product,
    shippingByProductId: {
      fakeId: {
        shipping: {
          price: null,
          currency: 'EUR',
        },
      },
    },
  },
};
