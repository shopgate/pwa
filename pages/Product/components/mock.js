import {
  basicProductState as bpState,
  productRouteMock,
  getShippingStateForId,
} from '@shopgate/pwa-common-commerce/product/mock';
import { getReviewsStateForId } from '@shopgate/pwa-common-commerce/reviews/mock';

const { productId } = 913;

const product = {
  ...bpState.product,
  ...getShippingStateForId(productId),
};
const basicProductState = {
  ...bpState,
  ...getReviewsStateForId(productId),
  product,
  router: {
    routing: false,
    stack: [{ ...productRouteMock }],
  },
  favorites: {
    products: {
      lastChange: 0,
      lastFetch: 0,
      expires: 99999999999999,
      ids: [],
      isFetching: false,
      ready: true,
    },
  },
};

export { basicProductState };
