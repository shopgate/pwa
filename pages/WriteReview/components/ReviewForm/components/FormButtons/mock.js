import { reviewRouteMock } from '@shopgate/pwa-common-commerce/reviews/mock';
/**
 * Mocked state with product only. Reviews not fetched.
 * @type {Object}
 */
export const mockedState = {
  history: {
    pathname: '/item/666f6f/write_review',
  },
  view: {
    isLoading: false,
  },
  router: {
    routing: false,
    stack: [
      reviewRouteMock,
    ],
  },
};
