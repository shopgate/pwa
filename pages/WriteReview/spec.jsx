import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import mockRenderOptions from '@shopgate/pwa-common/helpers/mocks/mockRenderOptions';
// eslint-disable-next-line import/named
import { mockedState } from './mock';

jest.mock('@shopgate/engage/components');
jest.mock('./components/ReviewForm', () => () => '');
const mockedStore = configureStore();
/**
 * Creates component
 * @return {ReactWrapper}
 */
const createComponent = () => {
  /* eslint-disable global-require */
  const { UnwrappedWriteReview } = require('./index');
  /* eslint-enable global-require */
  return mount(
    <Provider store={mockedStore(mockedState)}>
      <UnwrappedWriteReview productId="foo" />
    </Provider>,
    mockRenderOptions
  );
};

describe.skip('<WriteReview> page', () => {
  it('should not crash', () => {
    const component = createComponent();
    expect(component).toMatchSnapshot();
    expect(component.find('WriteReview').exists()).toBe(true);
  });
});
