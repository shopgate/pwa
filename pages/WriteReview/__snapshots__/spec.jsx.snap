// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WriteReview> page should not crash 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <WriteReview
    productId="foo"
  >
    <MockedView>
      <div>
        <Connect(BackBar)
          right={null}
          title="titles.reviews"
        >
          <BackBar
            goBack={[Function]}
            right={null}
            title="titles.reviews"
          >
            <AppBarDefault
              left={[Function]}
              right={null}
              title="titles.reviews"
            >
              <AppBar
                backgroundColor="#fff"
                below={
                  <React.Fragment>
                    <_default />
                  </React.Fragment>
                }
                center={
                  <AppBarTitle
                    title="titles.reviews"
                  />
                }
                left={[Function]}
                right={null}
                shadow={true}
                textColor="#000"
                title="titles.reviews"
              >
                <header
                  className={
                    Object {
                      "data-css-1gx0ht2": "",
                    }
                  }
                  data-test-id="Navigator"
                  role="banner"
                  style={
                    Object {
                      "background": "#fff",
                      "boxShadow": "rgba(0, 0, 0, .117647) 0 1px 6px, rgba(0, 0, 0, .117647) 0 1px 4px",
                      "color": "#000",
                    }
                  }
                >
                  <div
                    className={
                      Object {
                        "data-css-1qz3sla": "",
                      }
                    }
                  >
                    <AppBarTitle
                      title="titles.reviews"
                    >
                      <div
                        className={
                          Object {
                            "data-css-16ouezv": "",
                          }
                        }
                      >
                        titles.reviews
                      </div>
                    </AppBarTitle>
                  </div>
                  <_default>
                    <Consumer>
                      <ProgressBar
                        pattern=""
                        visible={true}
                      >
                        <ProgressBar
                          isVisible={false}
                        >
                          <Transition
                            appear={false}
                            enter={true}
                            exit={true}
                            in={false}
                            mountOnEnter={false}
                            onEnter={[Function]}
                            onEntered={[Function]}
                            onEntering={[Function]}
                            onExit={[Function]}
                            onExited={[Function]}
                            onExiting={[Function]}
                            timeout={150}
                            unmountOnExit={false}
                          >
                            <div
                              className="css-guf25f"
                              style={
                                Object {
                                  "transform": "scale(1, 0)",
                                }
                              }
                            >
                              <div
                                className="css-cd7tja"
                              />
                            </div>
                          </Transition>
                        </ProgressBar>
                      </ProgressBar>
                    </Consumer>
                  </_default>
                </header>
              </AppBar>
            </AppBarDefault>
          </BackBar>
        </Connect(BackBar)>
        <Component
          productId="foo"
        />
      </div>
    </MockedView>
  </WriteReview>
</Provider>
`;
